diff -uraN a/src/main.c b/src/main.c
--- a/src/main.c	2023-12-13 23:20:03.224036762 +0500
+++ b/src/main.c	2023-12-13 23:24:41.650731169 +0500
@@ -958,20 +958,8 @@
     //TODO: add relative path to box86 location
 #endif
 #ifndef TERMUX
-    if(FileExist("/lib/box86", 0))
-        AddPath("/lib/box86", &context->box86_ld_lib, 1);
-    if(FileExist("/usr/lib/box86", 0))
-        AddPath("/usr/lib/box86", &context->box86_ld_lib, 1);
-    if(FileExist("/lib/i386-linux-gnu", 0))
-        AddPath("/lib/i386-linux-gnu", &context->box86_ld_lib, 1);
-    if(FileExist("/usr/lib/i386-linux-gnu", 0))
-        AddPath("/usr/lib/i386-linux-gnu", &context->box86_ld_lib, 1);
-    if(FileExist("/lib/i686-pc-linux-gnu", 0))
-        AddPath("/lib/i686-pc-linux-gnu", &context->box86_ld_lib, 1);
-    if(FileExist("/usr/lib/i686-pc-linux-gnu", 0))
-        AddPath("/usr/lib/i686-pc-linux-gnu", &context->box86_ld_lib, 1);
-    if(FileExist("/usr/lib32", 0))
-        AddPath("/usr/lib32", &context->box86_ld_lib, 1);
+    if(FileExist("/data/data/com.termux/files/usr/glibc/lib32/i386-linux-gnu", 0))
+        AddPath("/data/data/com.termux/files/usr/glibc/lib32/i386-linux-gnu", &context->box86_ld_lib, 1);
 #else
     if(FileExist("/data/data/com.termux/files/usr/lib/i386-linux-gnu", 0))
         AddPath("/data/data/com.termux/files/usr/lib/i386-linux-gnu", &context->box86_ld_lib, 1);
@@ -1188,8 +1176,8 @@
 static void load_rcfiles()
 {
     #ifndef TERMUX
-    if(FileExist("/etc/box86.box86rc", IS_FILE))
-        LoadRCFile("/etc/box86.box86rc");
+    if(FileExist("/data/data/com.termux/files/usr/glibc/etc/box86.box86rc", IS_FILE))
+        LoadRCFile("/data/data/com.termux/files/usr/glibc/etc/box86.box86rc");
     #else
     if(FileExist("/data/data/com.termux/files/usr/etc/box86.box86rc", IS_FILE))
         LoadRCFile("/data/data/com.termux/files/usr/etc/box86.box86rc");
diff -uraN a/src/tools/fileutils.c b/src/tools/fileutils.c
--- a/src/tools/fileutils.c	2023-12-13 23:20:03.224036762 +0500
+++ b/src/tools/fileutils.c	2023-12-13 23:25:42.957403943 +0500
@@ -126,11 +126,8 @@
     if ((tmpdir = getenv ("TMPDIR")) != NULL) return tmpdir;
     if ((tmpdir = getenv ("TEMP")) != NULL)   return tmpdir;
     if ((tmpdir = getenv ("TMP")) != NULL)    return tmpdir;
-    if(FileExist("/tmp", 0))                  return "/tmp";
-    if(FileExist("/var/tmp", 0))              return "/var/tmp";
-    if(FileExist("/usr/tmp", 0))              return "/usr/tmp";
 
-    return "/tmp";  // meh...
+    return "/data/data/com.termux/files/usr/tmp";  // meh...
 }
 
 char* LowerCase(const char* s) {
