diff --git a/dlls/ntdll/unix/server.c b/dlls/ntdll/unix/server.c
index 53fcc61ccf3..051c62989b6 100644
--- a/dlls/ntdll/unix/server.c
+++ b/dlls/ntdll/unix/server.c
@@ -1260,7 +1260,7 @@ static const char *init_server_dir( dev_t dev, ino_t ino )
 #ifdef __ANDROID__  /* there's no /tmp dir on Android */
     asprintf( &dir, "%s/.wineserver/server-%s", config_dir, tmp );
 #else
-    asprintf( &dir, "/tmp/.wine-%u/server-%s", getuid(), tmp );
+    asprintf( &dir, "/data/data/com.termux/files/usr/tmp/.wine-%u/server-%s", getuid(), tmp );
 #endif
     return dir;
 }
@@ -1303,7 +1303,7 @@ static int setup_config_dir(void)
     {
         mkdir( "drive_c", 0777 );
         symlink( "../drive_c", "dosdevices/c:" );
-        symlink( "/", "dosdevices/z:" );
+        symlink( "/data/data/com.termux/files", "dosdevices/z:" );
     }
     else if (errno != EEXIST) fatal_perror( "cannot create %s/dosdevices", config_dir );

diff --git a/dlls/win32u/driver.c b/dlls/win32u/driver.c
index bc3409a9e34..833d5f33817 100644
--- a/dlls/win32u/driver.c
+++ b/dlls/win32u/driver.c
@@ -825,16 +825,16 @@ static BOOL nulldrv_CreateDesktop( const WCHAR *name, UINT width, UINT height )

 static BOOL nodrv_CreateWindow( HWND hwnd )
 {
-    static int warned;
-    HWND parent = NtUserGetAncestor( hwnd, GA_PARENT );
-
-    /* HWND_MESSAGE windows don't need a graphics driver */
-    if (!parent || parent == UlongToHandle( NtUserGetThreadInfo()->msg_window )) return TRUE;
-    if (warned++) return FALSE;
-
-    ERR_(winediag)( "Application tried to create a window, but no driver could be loaded.\n" );
-    if (driver_load_error[0]) ERR_(winediag)( "%s\n", debugstr_w(driver_load_error) );
-    return FALSE;
+//    static int warned;
+//    HWND parent = NtUserGetAncestor( hwnd, GA_PARENT );
+//
+//    /* HWND_MESSAGE windows don't need a graphics driver */
+//    if (!parent || parent == UlongToHandle( NtUserGetThreadInfo()->msg_window )) return TRUE;
+//    if (warned++) return FALSE;
+//
+//    ERR_(winediag)( "Application tried to create a window, but no driver could be loaded.\n" );
+//    if (driver_load_error[0]) ERR_(winediag)( "%s\n", debugstr_w(driver_load_error) );
+    return TRUE;
 }

 static BOOL nulldrv_CreateWindow( HWND hwnd )
diff --git a/programs/winebrowser/main.c b/programs/winebrowser/main.c
index 24416070152..67cac7fe1df 100644
--- a/programs/winebrowser/main.c
+++ b/programs/winebrowser/main.c
@@ -116,7 +116,7 @@ static int open_http_url( const WCHAR *url )
 {
     static const WCHAR defaultbrowsers[] =
         L"xdg-open\0"
-        "/usr/bin/open\0"
+        "/data/data/com.termux/files/usr/glibc/bin/open\0"
         "firefox\0"
         "konqueror\0"
         "mozilla\0"
@@ -141,7 +141,7 @@ static int open_http_url( const WCHAR *url )
 static int open_mailto_url( const WCHAR *url )
 {
     static const WCHAR defaultmailers[] =
-        L"/usr/bin/open\0"
+        L"/data/data/com.termux/files/usr/glibc/bin/open\0"
         "xdg-email\0"
         "mozilla-thunderbird\0"
         "thunderbird\0"
diff --git a/programs/winemenubuilder/winemenubuilder.c b/programs/winemenubuilder/winemenubuilder.c
index 36569fa9d5f..3583ef01bc6 100644
--- a/programs/winemenubuilder/winemenubuilder.c
+++ b/programs/winemenubuilder/winemenubuilder.c
@@ -1724,7 +1724,7 @@ static BOOL build_native_mime_types(struct list *mime_types)
     if (_wgetenv( L"XDG_DATA_DIRS" ))
         dirs = xwcsdup( _wgetenv( L"XDG_DATA_DIRS" ));
     else
-        dirs = xwcsdup( L"/usr/local/share/:/usr/share/" );
+        dirs = xwcsdup( L"/data/data/com.termux/files/usr/glibc/local/share/:/data/data/com.termux/files/usr/glibc/share/" );

     ret = add_mimes(xdg_data_dir, mime_types);
     if (ret)
diff --git a/server/request.c b/server/request.c
index 7021741c765..5a4a0dceaea 100644
--- a/server/request.c
+++ b/server/request.c
@@ -663,9 +663,9 @@ static char *create_server_dir( int force )
     strcpy( server_dir, config_dir );
     strcat( server_dir, "/.wineserver" );
 #else
-    len += sizeof("/tmp/.wine-") + 12;
+    len += sizeof("/data/data/com.termux/files/usr/tmp/.wine-") + 12;
     if (!(server_dir = malloc( len ))) fatal_error( "out of memory\n" );
-    sprintf( server_dir, "/tmp/.wine-%u", getuid() );
+    sprintf( server_dir, "/data/data/com.termux/files/usr/tmp/.wine-%u", getuid() );
 #endif
     create_dir( server_dir, &st2 );

diff --git a/server/unicode.c b/server/unicode.c
index f84520580d7..035be76394e 100644
--- a/server/unicode.c
+++ b/server/unicode.c
@@ -277,7 +277,7 @@ static char *get_nls_dir(void)
 /* load the case mapping table */
 struct fd *load_intl_file(void)
 {
-    static const char *nls_dirs[] = { NULL, NLSDIR, "/usr/local/share/wine/nls", "/usr/share/wine/nls" };
+    static const char *nls_dirs[] = { NULL, NLSDIR, "/data/data/com.termux/files/usr/glibc/local/share/wine/nls", "/data/data/com.termux/files/usr/glibc/share/wine/nls" };
     static const WCHAR nt_pathW[] = {'C',':','\\','w','i','n','d','o','w','s','\\',
         's','y','s','t','e','m','3','2','\\','l','_','i','n','t','l','.','n','l','s',0};
     static const struct unicode_str nt_name = { nt_pathW, sizeof(nt_pathW) };
